packages/devtools_app/lib/src/screens/vm_developer/vm_developer_common_widgets.dart

```diff
@@ -100,22 +105,357 @@ class VMInfoList extends StatelessWidget {
       ],
     );
   }
+}
 
-  List<Widget> _prettyRows(BuildContext context, List<Row> rows) {
-    return [
-      for (int i = 0; i < rows.length; ++i)
-        _buildAlternatingRow(context, i, rows[i]),
-    ];
+List<Widget> _prettyRows(BuildContext context, List<Row> rows) {
+  return [
+    for (int i = 0; i < rows.length; ++i)
+      _buildAlternatingRow(context, i, rows[i]),
+  ];
+}
+
+Widget _buildAlternatingRow(BuildContext context, int index, Widget row) {
+  return Container(
+    color: alternatingColorForIndex(index, Theme.of(context).colorScheme),
+    height: defaultRowHeight,
+    padding: const EdgeInsets.symmetric(
+      horizontal: defaultSpacing,
+    ),
+    child: row,
+  );
+}
+
+/// A Refresh ToolbarAction button.
+class ToolbarRefresh extends ToolbarAction {
+  const ToolbarRefresh({
+    super.icon = Icons.refresh,
+    required super.onPressed,
+    super.tooltip = 'Refresh',
+  });
+}
+
+/// An IconLabelButton with label 'Request' and a 'call made' icon.
+class RequestDataButton extends IconLabelButton {
+  const RequestDataButton({
+    required super.onPressed,
+    super.icon = Icons.call_made,
+    super.label = 'Request',
+    super.outlined = false,
+  });
+}
+
+/// Displays a RequestDataButton if [requestedSize] is null, otherwise displays
+/// the requestable size and a ToolbarRefresh button next to it,
+/// to request that size again if required.
+class RequestableSizeWidget extends StatelessWidget {
+  const RequestableSizeWidget({
+    required this.requestedSize,
+    required this.requestFunction,
+  });
+
+  final InstanceRef? requestedSize;
+  final void Function()? requestFunction;
```

Comment:
should this be non-nullable?

Reply:
Yes for the requestFunction, the requestedSize can be null.
I thought of leaving the requestFunction nullable, but I guess it doesn't make sense to use the widget without it. Thanks!